instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CoreRerouteTest.kt

```diff
@@ -288,4 +291,27 @@ class CoreRerouteTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.jav
             rerouteStates
         )
     }
+
+    @Test
+    fun reroute_to_alternative_route() = sdkTest {
+        val routes = RoutesProvider.dc_short_with_alternative(activity).toNavigationRoutes()
+        val origin = routes.first().routeOptions.coordinatesList().first()
+        mockLocationUpdatesRule.pushLocationUpdate {
+            latitude = origin.latitude()
+            longitude = origin.longitude()
+        }
+        mapboxNavigation.startTripSession()
+        mapboxNavigation.setNavigationRoutes(routes)
+        mapboxNavigation.routesUpdates().first { it.navigationRoutes == routes }
+
+        mockLocationReplayerRule.playRoute(routes[1].directionsRoute)
+        val routesUpdate = mapboxNavigation.routesUpdates().first {
+            it.reason == RoutesExtra.ROUTES_UPDATE_REASON_REROUTE
+        }
+
+        assertEquals(
+            listOf(routes[1]),
```

Comment:
Does this check verify that the request was not made?

Reply:
it verifies that alternative route has become primary

Reply:
But you say:
> We had a bug where navigator did reroute with new route request instead of switching to alternative.

If that were the case, the alternative would not have become primary?

Reply:
after full reroute, where navigation rerequest new set of routes from current position, primary route will be different, it won't match existing alternative